### ISSUE TITLE
deadvars tactic sometimes fail for unknown reason

### ISSUE THREAD

**Issue by LukeXuan** [PrincetonUniversity/VST#277]
`deadvars` tactic is failing sometimes. The case I met is caused by `find_dead_vars` not returning the correct list, which contains duplicates dead variables and live variables.

I will investigate later on an example that triggers the problem.

**Comment by LukeXuan** [comment]
I'm a little bit confused with the code in `deadvars.v`. On line 130, the function `deadvars_stmt` case if statement. after several calls, the variable are: `live'' := []`, `dead'' := [183%positive]`, `vl' := [183%positive]`. While the call to `deadvars_delete` leaves the `vl''` and `live3` the same as `vl'` and `live''`.

The call to `cont` just append the `vl''` to the dead variables, which is reasonable since it's the end of the function. However, the double appearance of `183%positive` makes it cause error when calling `drop` tactics.

I'm not sure which part of the call is incorrect, I suppose it's the call to `deadvars_delete` has wrong arguments?

**Comment by andrew-appel** [comment]
Can you see if this modification solves the problem?

```
--- a/floyd/deadvars.v
+++ b/floyd/deadvars.v
@@ -133,7 +133,8 @@ Fixpoint deadvars_stmt (vl: list ident) (live dead: list ident) (c: statement)
             let live'' := deadvars_union live1 live2 in
             let dead'' := deadvars_intersection dead1 dead2 in
             let (vl'',live3) := deadvars_delete live'' vl' live' in
-            cont vl'' live3 dead''
+            let (vl3, _) := deadvars_delete dead'' vl'' nil in
+            cont vl3 live3 dead''
      else
            let (vl', live')  := deadvars_removel [e] vl live in
             let (live1,dead1) := deadvars_stmt vl' live' dead c1 cont bcont in
```

**Comment by LukeXuan** [comment]
Yes it works now.

**Comment by andrew-appel** [comment]
Fixed in cd82c48.
